File: problems/problem_15.py
def main():
    print(count(20))

def count(squares):
    return int(factorial(squares*2) / factorial(squares)**2)

def factorial(number):
    factorial = 1
    for i in range(1, number+1):
        factorial *= i
    return factorial

if __name__ == "__main__":
    main()

File: problems/problem_15.rb
class LatticePaths

  private

  attr_writer :squares

  def initialize(squares)
    self.squares = squares
  end

  public

  attr_reader :squares

  def count
    factorial(squares*2) / factorial(squares)**2
  end

  def factorial(number)
    (1..number).inject(:*)
  end

end

if __FILE__ == $PROGRAM_NAME
  puts LatticePaths.new(20).count
end
